### PR TITLE
Make clouddriver caches configurable

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt [hunk 1]
```diff
@@ -50,59 +50,62 @@ import java.util.function.BiFunction
...
-  private val meterRegistry: MeterRegistry
+  private val meterRegistry: MeterRegistry,
+  cacheProperties: CacheProperties = CacheProperties()
 ) : CloudDriverCache {
 
   private val securityGroupSummariesByIdOrName = Caffeine.newBuilder()
     .executor(IO.asExecutor())
-    .maximumSize(1000)
-    .expireAfterWrite(1, MINUTES)
+    .maximumSize(cacheProperties.caches["securityGroups"]?.maximumSize ?: 1000)
```
Comment (lorin) [inline keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt:59]: At some point, I think we should introduce a helper method to DRY up the instantiation of these caches.

**Review by lorin — APPROVED** [review]
LGTM

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt [hunk 2]
```diff
@@ -50,59 +50,62 @@ import java.util.function.BiFunction
...
     .recordStats()
     .buildAsync<String, SecurityGroupSummary>()
     .withMetrics("securityGroups")
 
   private val networks = Caffeine.newBuilder()
     .executor(IO.asExecutor())
-    .maximumSize(1000)
-    .expireAfterWrite(1, HOURS)
+    .maximumSize(cacheProperties.caches["networks"]?.maximumSize ?: 1000)
+    .expireAfterWrite(cacheProperties.caches["networks"]?.expireAfterWrite ?: Duration.ofHours(1))
```
Comment (robfletcher) [inline keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt:68]: all the caches can be configured if necessary

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt [hunk 3]
```diff
@@ -50,59 +50,62 @@ import java.util.function.BiFunction
...
+    .maximumSize(cacheProperties.caches["subnetsByPurpose"]?.maximumSize ?: 1000)
+    .expireAfterWrite(cacheProperties.caches["subnetsByPurpose"]?.expireAfterWrite ?: Duration.ofHours(1))
     .recordStats()
     .buildAsync<String, Subnet>()
     .withMetrics("subnetsByPurpose")
 
   private val namedImagesByAccountAndImageName = Caffeine.newBuilder()
     .executor(IO.asExecutor())
+    .maximumSize(cacheProperties.caches["namedImages"]?.maximumSize ?: 1000)
+    .expireAfterWrite(cacheProperties.caches["namedImages"]?.expireAfterWrite ?: Duration.ofMinutes(5))
```
Comment (robfletcher) [inline keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt:108]: Set a TTL / max size on the AMI cache

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt [hunk 4]
```diff
@@ -50,59 +50,62 @@ import java.util.function.BiFunction
...
+  private val meterRegistry: MeterRegistry,
+  cacheProperties: CacheProperties = CacheProperties()
 ) : CloudDriverCache {
 
   private val securityGroupSummariesByIdOrName = Caffeine.newBuilder()
     .executor(IO.asExecutor())
-    .maximumSize(1000)
-    .expireAfterWrite(1, MINUTES)
+    .maximumSize(cacheProperties.caches["securityGroups"]?.maximumSize ?: 1000)
+    .expireAfterWrite(cacheProperties.caches["securityGroups"]?.expireAfterWrite ?: Duration.ofMinutes(10))
```
Comment (robfletcher) [inline keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt:60]: Extended the default here

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt (hunk 1 above)
Comment (robfletcher) [reply keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt]: Done!

**Pull request merged**